07_safety/juvenile-arrests/juvenile-arrests-county-all.Rmd

```diff
@@ -0,0 +1,971 @@
+    ethn_nonhisp = sum(ethnicity_of_arrestee == "not of hispanic origin"),
+    # Sex subgroups
+    sex_male = sum(sex_of_arrestee == "male"),
+    sex_female = sum(sex_of_arrestee == "female"),
+    # Age subgroups
+    age_10_14 = sum(age >= 10 & age <= 14),
+    age_15_17 = sum(age >= 15 & age <= 17),
+    # Aggregating group as well
+    group_a = max(group=="A"),
+    group_b = max(group=="B")
+  ) %>%
+  ungroup() %>%
+  # Construct last race/ethnicity subgroup
+  mutate(race_asian_other = race_aian + race_asian + race_nhpi)
+
+```
+
+1,458 agencies across all years only have Group B arrests of juveniles
+```{r}
+arrests_agency %>% 
+  count(group_a, group_b)
+
+```
+
+Make narrow version with just needed variables
+```{r}
+arrests_agency <- arrests_agency %>%
+  select(
+    year,
+    ori, 
+    arr_total, 
+    # arr_viol, 
+    # arr_prop, 
+    starts_with("race"), 
+    ethn_hispanic,
+    starts_with("sex"),
+    starts_with("age")
+  )
+
+```
+
+
+## 4. Aggregate agency-level arrests to the county level
+This section aggregates the agency-level arrest counts from the agency-level to the county-level.
+
+Our data are currently long by year/agency. However, we want to know the arrest counts at the county level, not the agency level. In the `01_agency_geo_county.Rmd` program, we constructed a file that linked reporting agencies to counties. For agencies that served more than one county, the file includes a weight to assign proportions of agency-level counts to each county based on population. We can use that file to aggregate counts from agencies to counties.
+
+### 4.1 Load universe of county/agency pairs with weights
+First load the universe of agencies linked to counties with weight for each county. This file was created in `01_agency_geo_county.Rmd` and is long by unique agency/county pairs.
+```{r}
+agency_weights <- read_csv(here::here("07_safety", "modified data", "all_agency_county.csv")) 
+
+```
+
+### 4.2 Join agency-level arrests for all years onto the county-agency file
+Join arrests onto the full universe of county-agencies and multiply counts by the agency-county weight.
+```{r}
+# Join agency-level juvenile arrests onto universe of agencies
+arrests_agency_county <- left_join(
+  x = agency_weights,
+  y = arrests_agency, 
+  by = c("ori", "year")
+) %>%
+# Multiply stats by agency-county weight
+  mutate(across(arr_total:age_15_17, ~.x * weight))
+
+```
+
+There are a lot of unmatched observations. 23,948 observations are only in our universe of agencies with county-agency weights. My first thought is that since this is the universe of *all* agencies, some agencies may not have reported arrests and are therefore not in the arrests data. 1,127 observations are only in the 2021-2022 agency-level arrests data. My first thought is that since we only used the Batch Header File from 2022 to construct the universe of agencies and agency-county weights, it may be missing agencies from other years. 
```

Comment:
I'm still seeing 23,948 not matched. In my next push, I will make sure you have the right `all_agency_county.csv` file that we are joining onto and that was created in `01_agency_geo_county.qmd`. Maybe I accidentally pushed an old version?

Also, that is a great point about us limiting to arrests under 18. I included that in the text.

Reply:
Just flagging that I re-ran the `01_agency_geo_county.qmd` file to make sure that the most recent versions of the files created in that program are available. I think what might have happened is that, because `01_agency_geo_county.qmd` and `01_agency_geo_place.qmd` are identical for both the `reported-crime` and `juvenile-arrests` metrics, I would just copy them from one branch to the other whenever I made changes to those files. But, for the files they create (e.g., `agency_weights` in this section) to update on the other branch, they also need to be ran on that branch. I think at one point I just copied the `01_agency_geo` programs without remembering to run them, so this branch had older versions of the files created by those programs.

Reply:
Amy, try running `01_agency_geo_county.qmd` again on this branch and see if the numbers still aren't matching. That program creates the file joined in this step, and I think you might have been working with an older version.